File: SpacifyAPI/SpacifyAPI/Models/Responses/ConferenceRoomResponse.cs
using SpacifyAPI.Entities;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace SpacifyAPI.Models.Responses
{
    public class ConferenceRoomResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string EquipmentDetails { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public int Capacity { get; set; }
        public int FloorId { get; set; }

        public List<ConferenceRoomReservationResponse>? ConferenceRoomReservations { get; set; } = new List<ConferenceRoomReservationResponse>();

    }
}

File: SpacifyFrontend/src/app/models/response/conference-room-response.ts
import { ConferenceRoomReservationResponse } from './conference-room-reservation-response';

export interface ConferenceRoomResponse {
  id: number;
  name: string;
  equipmentDetails: string;
  imageUrl: string;
  capacity: number;
  floorId: number;
  conferenceRoomReservations?: ConferenceRoomReservationResponse[];
}
